pwiz_tools/Skyline/SkylineFiles.cs

```diff
@@ -285,6 +285,23 @@ public bool OpenFile(string path, FormEx parentWindow = null)
             Exception exception = null;
             SrmDocument document = null;
 
+            // A fairly common support question is "why won't this Skyline file open?" when they are actually
+            // trying to open a .skyd file or somesuch.  Probably an artifact of Windows hiding file extensions.
+            // Try to work around it by finding a plausible matching .sky file when asked to open a .sky? file.
+            if (!path.EndsWith(SrmDocument.EXT))
+            {
+                var index = path.LastIndexOf(SrmDocument.EXT, StringComparison.Ordinal);
+                if (index > 0 && index == path.Length - (SrmDocument.EXT.Length + 1))
+                {
+                    // Looks like user picked a .skyd or .skyr or .skyl etc
+                    var likelyPath = path.Substring(0, index + SrmDocument.EXT.Length);
+                    if (File.Exists(likelyPath))
+                    {
+                        path = likelyPath;

```

Comment:
Should we also throw a nicer error, if the .sky file does not exist but they have chosen a .skyd, .skyl, .skyr?

Reply:
Yeah, we should. Like this:
![image](https://user-images.githubusercontent.com/1200761/55513929-bc041280-561b-11e9-9f47-955e85c457bb.png)

Reply:
Not sure about the "and contain XML data". I would say "in XML format" but that is not true of a .sky.zip file. So maybe just leave it at the extensions expected.

Reply:
Can we go with "in XML format"? I wouldn't want to risk giving the impression that it was just the .ext that we were concerned with. And I was thinking that the user would understand that .zip isn't literally xml data, but that we expected to find xml data inside.

Reply:
Okay.

Reply:
I mean, I can just imagine somebody renaming their .raw file to .sky in a frantic attempt to shoehorn it into Skyline without the bother of learning how to use it first.

Reply:
Approved, then?